plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,590 @@
+        if share_type == "local":
+            export_license_status = export_license_local(self, module)
+        elif share_type in ["http", "https"]:
+            export_license_status = export_license_http(self, module)
+            job_status = get_job_status(self, module, export_license_status)
+        elif share_type == "cifs":
+            export_license_status = export_license_cifs(self, module)
+            job_status = get_job_status(self, module, export_license_status)
+        elif share_type == "nfs":
+            export_license_status = export_license_nfs(self, module)
+            job_status = get_job_status(self, module, export_license_status)
+        status = export_license_status.status_code
+        if share_type in ["http", "https", "cifs", "nfs"]:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_EXPORT_MSG, changed=True, job_details=job_status)
+            else:
+                module.exit_json(msg=FAILURE_MSG.format(operation="export", license_id=license_id), changed=False, job_details=job_status)
+        else:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_EXPORT_MSG, changed=True, job_details={})
+            else:
+                module.exit_json(msg=FAILURE_MSG.format(operation="export", license_id=license_id), changed=False, job_details={})
+        return export_license_status
+
+
+def check_license_id(self, module, license_id, operation):
+    try:
+        response = self.idrac.invoke_request(LICENSE_URI.format(license_id=license_id), 'GET')
+        return response
+    except Exception:
+        module.exit_json(msg=FAILURE_MSG.format(operation=operation, license_id=license_id), changed=False)
+
+
+def get_job_status(self, module, export_license_status):
+    res_uri = validate_and_get_first_resource_id_uri(self.module, self.idrac, SYSTEMS_URI)
+    job_tracking_uri = export_license_status.headers.get("Location")
+    job_id = job_tracking_uri.split("/")[-1]
+    job_uri = IDRAC_JOB_URI.format(job_id=job_id, res_uri=res_uri[0])
+    job_failed, msg, job_dict, wait_time = idrac_redfish_job_tracking(self.idrac, job_uri)
+    job_dict = remove_key(job_dict, regex_pattern='(.*?)@odata')
+    if job_failed:
+        module.exit_json(
+            msg=job_dict.get('Message'),
+            changed=False,
+            job_details=job_dict)
+    return job_dict
+
+
+def export_license_local(self, module):
+    uri = validate_and_get_first_resource_id_uri(
+        self.module, self.idrac, SYSTEMS_URI)
+    export_license_url = f"{uri[0]}/Oem/Dell/DellLicenseManagementService/Actions/DellLicenseManagementService.ExportLicense"
+    payload = {}
+    payload["EntitlementID"] = module.params.get('license_id')
+    path = module.params.get('share_parameters').get('share_name')
+    if not (os.path.exists(path) or os.path.isdir(path)):
+        module.exit_json(msg=f"Provided directory path '{path}' is not valid.", failed=True)
```

Comment:
messages better to move it to top instead of inline.

Reply:
Done